pkg/serverhandler/handlertest/test_service_logs.go

```diff
@@ -453,10 +436,19 @@ func TestServiceGetLogStream_depPlugin(t *testing.T) {
 			})
 		}
 
-		logSendClient.Send(&pb.EntrypointLogBatch{
-			InstanceId: instanceId,
-			Lines:      entries,
-		})
+		// Our instance log buffer might not be registered, so it's OK to retry
+		require.Eventually(t, func() bool {
+			err := logSendClient.Send(&pb.EntrypointLogBatch{
```

Comment:
This was the only real change required to make these tests consumable - we can't couple directly to the inmemstate serverstate, so we need a different way to make sure the buffer is initialized.

This is a little less precise of a check, but makes sense to me and seems to work in practice.